frida_tools/tracer.py

```diff
@@ -106,7 +110,10 @@ def on_trace_error(self, error):
         def on_trace_events(self, events):
             no_attributes = Style.RESET_ALL
             for timestamp, thread_id, depth, target_address, message in events:
-                if self._quiet:
+                if self._output is not None:
+                    with open(self._output, "a") as fh:
```

Comment:
won't this have the effect of constantly opening and closing the output for every event? that's a lot of syscalls isn't it?

Reply:
seems like it would be more efficient to open the file once outside the loop at least, or open it once at the class level and check that the handle is available before writing the message?